src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp

```diff
@@ -11,91 +11,120 @@ SDComment: Conflag on ground nyi
 SDCategory: Molten Core
 EndScriptData */
 
-#include "molten_core.h"
-#include "ObjectMgr.h"
-#include "ScriptedCreature.h"
 #include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "molten_core.h"
+#include "SpellScript.h"
 
 enum Texts
 {
-    EMOTE_FRENZY = 0,
-    EMOTE_SMOLDERING = 0,
-    EMOTE_IGNITE = 1,
+    EMOTE_FRENZY                    = 0,
 };
 
 enum Spells
 {
-    SPELL_FRENZY = 19451,
-    SPELL_MAGMA_SPIT = 19449,
-    SPELL_PANIC = 19408,
-    SPELL_LAVA_BOMB = 19428,
-    SPELL_SERRATED_BITE = 19771,
+    SPELL_FRENZY                    = 19451,
+    SPELL_MAGMA_SPIT                = 19449,
+    SPELL_PANIC                     = 19408,                    // Aoe fear
+    SPELL_LAVA_BOMB                 = 19411,                    // This calls a dummy server side effect that cast spell 20494 to spawn GO 177704 for 30s
+    SPELL_LAVA_BOMB_EFFECT          = 20494,                    // Spawns trap GO 177704 which triggers 19428
+    SPELL_LAVA_BOMB_RANGED          = 20474,                    // This calls a dummy server side effect that cast spell 20495 to spawn GO 177704 for 60s
+    SPELL_LAVA_BOMB_RANGED_EFFECT   = 20495,                    // Spawns trap GO 177704 which triggers 19428
 };
 
 enum Events
 {
-    EVENT_FRENZY = 1,
-    EVENT_PANIC = 2,
-    EVENT_LAVA_BOMB = 3,
-    EVENT_SERRATED_BITE = 1,
-    EVENT_IGNITE = 2,
+    EVENT_FRENZY                    = 1,
+    EVENT_PANIC,
+    EVENT_LAVA_BOMB,
+    EVENT_LAVA_BOMB_RANGED,
 };
 
+constexpr float MELEE_TARGET_LOOKUP_DIST = 10.0f;
```

Comment:
Is this the DBC value of melee rnage?

Reply:
This is just a constant we use to differentiate between the ranged and meele lava bomb. We experimented with different values and 10 yards looked like the most blizzlike value.